docs/admin-guide/provisioning/vsphere.md

```diff
@@ -64,7 +66,68 @@ karina ca generate --name root-ca --cert-path .certs/root-ca.crt --private-key-p
 karina ca generate --name ingress-ca --cert-path .certs/ingress-ca.crt --private-key-path .certs/ingress-ca.key --password foobar
 ```
 
-## 4. Configure the platform config
+## 4. Setup Service discovery
+
+Karina requires a service discovery mechanism to facilitate the initial connection to the kubernetes hosts.  A containerised consul service discovery can be enabled on a host in the vsphere cluster using the [konfigadm](https://github.com/flanksource/konfigadm) tool:
+
+```bash
+konfigadm apply - << EOF
+commands:
+  - mkdir -p /opt/consul
+  - chown -R 100:1000 /opt/consul
+  - iptables -A PREROUTING -t nat -p tcp --dport 80 -j REDIRECT --to-ports 8500
+container_runtime:
+  type: docker
+containers:
+  - image: docker.io/consul:1.9.1
+    docker_opts: --net=host
+    args: agent -server -ui -data-dir /opt/consul -datacenter lab -bootstrap
+    volumes:
+      - /opt/consul:/opt/consul
+    env:
+      CONSUL_BIND_INTERFACE: ens160
+      CONSUL_CLIENT_INTERFACE: ens160
+EOF
+```
+
+## 5. Generate Template image
+
+To create a template image with the prerequisite configuration installed use the [konfigadm](https://github.com/flanksource/konfigadm) tool:
+
+```bash
+# Create config file
```

Comment:
It might make sense to style this as a  procedure e.g. 

1. Create a konfigadm spec
```
```
1. Build the image
```
```
1. Define GOVC variables
```bash
export GOVC_URL=...
....
```
Test that govc can connect via `govc about`
1. Upload the image
....

1. Configure the template etc..